Feature:Integrate Device Admin Permission and Refine Permission Flow

KvRae opened the pull request:
This pull request introduces a crucial upgrade to the app's blocking functionality by integrating **Device Administrator** permission and significantly refining the overall permission request process to improve the user experience.

---

##  Key Changes & Enhancements

### 1. Device Admin Integration for Enhanced Blocking

The core change is the introduction of the **Device Admin** permission to enhance the reliability and effectiveness of the app-blocking feature.

* **Added:** `AppDeviceAdminReceiver` and necessary XML configurations to register the app as a device administrator.
* **Implemented:** Utility functions (`hasAdminPermission`, `requestAdminPermission`) to check for and prompt for Device Admin privileges on app startup.
* **Updated:** UI strings in **English and Arabic** to clearly explain the need for the new administrative permission.
* **Modified:** Device Admin is now part of the initial permission request flow.

### 2. Streamlined Sequential Permission Handling

The entire permission request logic has been refactored to be more sequential and user-friendly, ensuring a smooth onboarding experience.

* **Refactored:** Permission handling logic now requests permissions sequentially *after* the user confirms they want to enable protection.
* **Process:** The app iterates through a predefined chain of required permissions (e.g., Usage Stats, Overlay, Accessibility, and the new Device Admin) and prompts the user for each missing one.
* **Benefit:** Once all necessary permissions are granted, protection is **automatically activated**, and a confirmation message is displayed, significantly streamlining the onboarding process.
* **Removed:** Usage Stats permission has been removed from the *initial* permission chain, but is now integrated into the new sequential flow.

### 3. UI and Configuration Updates

Minor structural and accessibility improvements have also been included:

* **Refactoring:** UI logic was moved from `AppActivity` into a new, cleaner **`MainScreen.kt` composable**.
* **Accessibility:** Updated the accessibility service configuration to correctly target the settings app and be explicitly set as an accessibility tool.

---

##  Testing Notes

Please verify the following:
1.  On a clean install, the app correctly prompts for the new **Device Admin** permission.
2.  The sequential flow prompts the user for *all* required permissions (Device Admin, Usage Stats, Overlay, Accessibility, etc.).
3.  Once all permissions are granted, protection is enabled automatically.
4.  The new **English and Arabic** strings for the administrative permission dialog are displayed correctly.

KvRae committed:
Update README.md

KvRae committed:
Feat: Ui Enhancement:
- Added splash screen to the app using the new splashscreen api
- Added dark theme support
- Enhanced SupportScreen.kt topbar navigation button
- Worked on code clean ups (moved hardcoded strings to strings.xml)
- replaced gradle dependencies imports using the new version catalog style
- fixed the items color on status bar text on light mode using the edge to edge feature

KvRae committed:
- Removed the AppViewModelTest.kt implementation since its not working yet

KvRae committed:
refactor: use string resources for localization

KvRae committed:
feat: force Arabic locale and add backup rules

- Add `locales_config.xml` and `data_extraction_rules.xml`.
- Implement `LocaleHelper` to force the app's language to Arabic.
- Update `AppActivity` to use the new locale helper.
- Correct a constant name from `vpnAddress` to `VPN_ADDRESS`.

KvRae committed:
style(support): add padding to support screen

KvRae committed:
Merge branch 'dev' into enhancement/ui-enhancement

KvRae committed:
feat: internationalize EnableProtectionDialog

- Replaced hardcoded strings in `EnableProtectionDialog.kt` with string resources.
- Added English and Arabic translations for the new strings.

KvRae committed:
refactor: implement dark theme and RTL support

- Update theme colors for better dark mode appearance.
- Apply MaterialTheme colors and typography across various UI components (`AppBlockDialog`, `EnableProtectionDialog`, `LockScreenOverlay`, `ScreenshotOverlay`).
- Change the app name in `strings.xml` to Arabic script.
- Force Right-to-Left (RTL) layout and Arabic locale for the `LockScreenOverlay`.

KvRae committed:
feat: Force light theme

Force the light theme by disabling the dark theme logic. The app will now use `dynamicLightColorScheme` on supported devices or default to `LightColorScheme`.

KvRae committed:
feat: Add Device Admin permission for app blocking

This commit introduces the Device Admin permission to enhance the app blocking functionality.

- Adds `AppDeviceAdminReceiver` and necessary XML configurations to register the app as a device administrator.
- Implements utility functions (`hasAdminPermission`, `requestAdminPermission`) to check for and request Device Admin privileges on startup.
- Includes a new string resource for the admin permission explanation.
- Refactors UI logic by moving it from `AppActivity` into a new `MainScreen.kt` composable.
- Updates the accessibility service configuration to target the settings app and be an accessibility tool.

KvRae committed:
feat: Implement sequential permission requests

Refactors the permission handling logic to request permissions sequentially. After the user confirms enabling protection, the app now iterates through a predefined chain of required permissions (Usage Stats, Overlay, Accessibility, etc.) and prompts the user for each one that is missing.

Once all necessary permissions are granted, protection is automatically activated, and a confirmation message is displayed. This streamlines the onboarding process for the user.

KvRae committed:
feat: Add Device Admin permission check

- Add Device Admin to the permission request flow.
- Remove UsageStats from the initial permission chain.
- Update UI strings for the new administrative permission dialog in English and Arabic.

KvRae committed:
Merge branch 'dev' into feature/DeviceAdminPolicy

KvRae committed:
feat: Make accessibility service persistent

- Run `MyAccessibilityService` as a foreground service with a persistent notification to prevent the system from killing it.
- Implement a mechanism to automatically restart the service if it's terminated.
- Refactor string resources for better maintainability.
- Fix text color issues on the support screen.

AmmarYasserAllaithy commented on app/src/main/res/values/strings.xml:
```diff
@@ -127,4 +127,9 @@
     <string name="classic_unistall_message2">Therefore, your agreement to enable protection is considered an explicit consent to activate the uninstall protection feature and to comply with all associated security restrictions.</string>
     <string name="enable_protection_confirm_text">Do you want to proceed and enable protection now?</string>
+    <string name="admin_permission_message">Please enable Device Admin to allow app blocking functionality.</string>
+    <string name="protection_activated_message">\n                       لحماية مفعلة الآن!\n                       يمكنك تعديل التطبيقات المحجوبة من خلال الضغط على \"حظر التطبيقات\" في الشاشة الرئيسية</string>
```
Is it intended NOT to translate these strings?

AmmarYasserAllaithy commented on app/src/main/res/xml/device_admin_receiver.xml:
Why all these policies? 
I think they are unrelated, Do we need any of them?

KvRae replied:
Forget it, I realized what you're targeting with the policies. I was attempting to create a scenario involving customer support, but it seems I misinterpreted the concept and neglected to eliminate them as well. So, thanks, I’ll take them out, and we’ll need to have a conversation about the scenario and how it will function on both mobile and backend.

AmmarYasserAllaithy commented on app/src/main/java/com/mafazaa/ainaa/MainScreen.kt:
This file is unused!

AmmarYasserAllaithy commented on app/src/main/java/com/mafazaa/ainaa/AppActivity.kt:
```diff
@@ -409,13 +429,52 @@ class AppActivity : ComponentActivity() {
         }
 
+    }
+    private fun findNextMissingPermission(): PermissionState? {
```
Great job 👏🏻 👏🏻

KvRae committed:
refactor: remove MainScreen.kt

The `MainScreen.kt` file, which contained the main UI composable and navigation logic, has been deleted.

KvRae committed:
Refactor: Change permission request order and clean up admin policies

KvRae committed:
feat: Add watchdog to ensure service persistence

Introduces a `WatchdogReceiver` to ensure the `MyAccessibilityService` remains active.

- The service now schedules a repeating alarm (`scheduleWatchdog`) when it starts. This alarm triggers the `WatchdogReceiver` periodically.
- The `WatchdogReceiver` checks if the service is running and restarts it if it's not.
- The alarm is canceled when the service is stopped (`cancelWatchdog`).
- The `BootReceiver` is updated to explicitly start the accessibility service on device boot.

KvRae committed:
chore: Remove redundant comment

Removes an unnecessary comment from `AndroidManifest.xml`.

KvRae committed:
feat: Increase watchdog interval and clean up service creation

Increased the watchdog interval from 30 seconds to 15 minutes.
Removed redundant `LockOverlayManager` initialization in `MyAccessibilityService`'s `onCreate` method and added a log message for service creation.

KvRae committed:
refactor: Start accessibility service on boot and inject LockOverlayManager

Starts MyAccessibilityService on device boot using `startForegroundService` in BootReceiver.
This ensures the service starts correctly on newer Android versions.

Injects `LockOverlayManager` in MyAccessibilityService's `onCreate` method.
Adds the `LOCKED_BOOT_COMPLETED` action to the BootReceiver's intent filter in the manifest to handle direct boot scenarios.

AmmarYasserAllaithy reviewed (approved):
Great work
